spec/lib/jobs/dell-wsman-bios-spec.js

```diff
@@ -0,0 +1,236 @@
+            }).catch(function(error){
+                expect(error.name).to.equal("NotFoundError");
+                expect(error.message).to.equal("Dell Configuration (BIOS) web service is not defined in smiConfig.json.");
+            });
+        });
+
+        it('Should finish with an error: No target IP address', function(){
+            var obms = {
+                "service" : "dell-wsman-obm-service",
+                "config" : {
+                    "user" : "admin",
+                    "password" : "admin",
+                },
+                "node" : "59db1dc1423ad2cc0650f8bc"
+            };
+            BaseJob.prototype.checkOBM.resolves(obms);
+            sandbox.stub(configuration,'get').returns(configFile);
+            BaseJob.prototype.getIpAddress.resolves(undefined);
+
+            return job._initJob().then(function(){
+                expect(job.inventories).to.equal("undefined");
+            }).catch(function(error){
+                expect(error.name).to.equal("NotFoundError");
+                expect(error.message).to.equal("No target IP address.");
+            });
+        });
+
+    });
+
+    describe('handle cases for _handleAsyncRequest function', function(){
+        it('Should send request to smi service successfully', function(){
+            job.wsman = new WsmanTool();
+            job.inventories = ['bios', 'boot'];
+            job.dellConfigs = {
+                "wsmanCallbackUri": "http://172.31.128.1:9988/api/2.0/wsmanCallback/_IDENTIFIER_",
+                "services": {
+                    "inventory": {
+                        "serverCallback": "/api/1.0/server/inventory/callback"
+                    }
+                }
+            };
+            var response = {
+                "body": {
+                    "response": "Request Submitted!"
+                }
+            };
+            WsmanTool.prototype.clientRequest.resolves(response);
+            return job._handleAsyncRequest().then(function(){
+                expect(job._handleAsyncRequest()).to.be.fulfilled;
+            });
+        });
+
+        it('Should fail to get respone from smi service', function(){
+            job.wsman = new WsmanTool();
+            job.inventories = ['bios', 'boot'];
+            job.dellConfigs = {
+                "wsmanCallbackUri": "http://172.31.128.1:9988/api/2.0/wsmanCallback/_IDENTIFIER_",
+                "services": {
+                    "inventory": {
+                        "serverCallback": "/api/1.0/server/inventory/callback"
+                    }
+                }
+            };
+            var response = {
+                "body": {
+                    "response": "Request response"
+                }
+            };
+            WsmanTool.prototype.clientRequest.resolves(response);
+            return job._handleAsyncRequest().then(function(){
+                expect(response.body.response.indexOf('Submitted')).to.equal(-1);
```

Comment:
It doesn't make sense here since `response` is not returned from `_handleAsyncRequest`.